mmdet/integration/nncf/runners.py

```diff
@@ -0,0 +1,76 @@
+import warnings
+import time
+
+import mmcv
+from mmcv.runner import EpochBasedRunner
+from mmcv.runner import RUNNERS
+from mmcv.runner.utils import get_host_info
+from mmdet.core.evaluation.eval_hooks import EvalHook
+
+from .utils import check_nncf_is_enabled
+
+
+@RUNNERS.register_module()
+class AccuracyAwareRunner(EpochBasedRunner):
+    """
+    An mmdet training runner to be used with NNCF-based accuracy-aware training.
+    Inherited from the standard EpochBasedRunner with the overridden "run" method.
+    This runner does not use the "workflow" and "max_epochs" parameters that are
+    used by the EpochBasedRunner since the training is controlled by NNCF's
+    AdaptiveCompressionTrainingLoop that does the scheduling of the compression-aware
+    training loop using the parameters specified in the "accuracy_aware_training".
+    """
+    def __init__(self, *args, target_metric_name='bbox_mAP', **kwargs):
+        super().__init__(*args, **kwargs)
+        if not any([isinstance(hook, EvalHook) for hook in self._hooks]):
+            raise RuntimeError('No evaluation hooks were registered for the training. '
+                               'Please register an EvalHook or a derived class hook object '
+                               'to enable epoch-based metric calculation.')
+        self.target_metric_name = target_metric_name
+
+    def run(self, data_loaders, *args, compression_ctrl=None,
+            nncf_config=None, configure_optimizers_fn=None):
```

Comment:
Please, note that probably it could be useful to add `**kwargs` to this method `run` (and ignore it).
I'm not sure that this is mandatory, but several other runners (like `EpochBasedRunner`) do it.